tutorial/readme.md

```diff
@@ -324,3 +324,66 @@ COPY ./.tmux.conf /tmp/.tmux.conf
 RUN cat /tmp/.tmux.conf > ~/.tmux.conf && \
 	sudo rm /tmp/.tmux.conf
 ```
+
+## 5. NodeJS
+I found the best way to install node and manage node versions is to use 
+[Node Version Manager (nvm)](https://github.com/creationix/nvm). First we need 
+to cURL the install script (wget if you did not install cURL), then we set the 
+`NVM_DIR` environment variable for docker and lastly we source nvm.sh, install
+the [node version](https://github.com/creationix/nvm#long-term-support) we want,
+in this case the LTS version.
+```dockerfile
+# get the nvm install script and run it
+RUN curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.33.11/install.sh | bash
+
+# set the environment variable
+ENV NVM_DIR $HOME/.nvm
+
+# source nvm, install the version we want, alias that version so it always loads
+RUN . "$NVM_DIR/nvm.sh" && nvm install --lts && nvm alias default stable
+```
+To test this build and run the container and try
+```bash
+node --version
+```
+If installed correctly you should the version number for node.
+
+### YouCompleteMe
+`YouCompleteMe` is a autocomplete tool. To bundle JS with `YouCompleteMe`
+you need to source `nvm.sh` and run the install script with `--js-completer`
+added on.
+```dockerfile
+RUN . "$NVM_DIR/nvm.sh" && \
+	python3 ~/.config/nvim/plugged/YouCompleteMe/install.py \
+	--js-completer
+```
+#### Adding Tern completion engine
+To turn on [Tern](http://ternjs.net/)
+we need a `.tern-config` file. It will contain
+a JSON object, on your host machine:
+```bash
+touch .tern-config
+```
+Edit `.tern-config` with your editor of choice
+```json
+{
+    "plugins": {
+        "node": {}
+    }
+}
+```
+More info can be found in the 
+[tern documentation](http://ternjs.net/doc/manual.html).
+Now we need to copy and own the file. In the Dockerfile:
+```dockerfile
+# copy tern completion config
+COPY ./.tern-config "$HOME/.tern-config"
```

Comment:
Wouldn't it be easier to just tell the person to run this:

```
RUN echo '{"plugins": {"node": {}}}' > ~/.ternconfig
```

Reply:
yes